content/docs/atom-guide/dev-guide/materials/lua-material-functor-api.md

```diff
@@ -0,0 +1,387 @@
+  * **GetMaterialPropertyValue_Vector4**(`string`): Returns a `Vector4` type. See [Lua Math Library](/docs/user-guide/scripting/lua/math-library/).
+  * **GetMaterialPropertyValue_Color**(`string`): Returns a `Color` type. See [Lua Math Library](/docs/user-guide/scripting/lua/math-library/).
+  * **GetMaterialPropertyValue_Image**(`string`): Returns a generic pointer `userdata` type if an image is available, or `nil` otherwise.
+
+### HasMaterialProperty(`string`)
+
+Returns a boolean whether a property with the given name exists. Note the material property does not have to be listed in [GetMaterialPropertyDependencies](#main-functions) because this functions is just checking availability, not *reading values*.
+
+### SetShaderConstant\_
+
+Each **SetShaderConstant\_** function takes a `string` shader input name and value to set. You must use the version that matches the data type of the shader input.
+
+  * **SetShaderConstant_bool**(`string`, `boolean`)
+  * **SetShaderConstant_int**(`string`, `number`)
+  * **SetShaderConstant_uint**(`string`, `number`)
+  * **SetShaderConstant_float**(`string`, `number`)
+  * **SetShaderConstant_Vector2**(`string`, `Vector2`)
+  * **SetShaderConstant_Vector3**(`string`, `Vector3`)
+  * **SetShaderConstant_Vector4**(`string`, `Vector4`)
+  * **SetShaderConstant_Color**(`string`, `Color`)
+  * **SetShaderConstant_Matrix3x3**(`string`, `Matrix3x3`)
+  * **SetShaderConstant_Matrix4x4**(`string`, `Matrix4x4`)
+
+### SetShaderOptionValue\_
+
+Each **SetShaderOptionValue\_** function takes a `string` shader option name and value to set. The value will be applied to all shaders in the material type that have an option with the given name. You must use the version that matches the data type of the shader option. The shader option must be listed in [GetShaderOptionDependencies](#main-functions).
+
+  * **SetShaderOptionValue_bool**(`string`, `boolean`)
+  * **SetShaderOptionValue_uint**(`string`, `number`)
+  * **SetShaderOptionValue_enum**(`string`, `number`)
+
+{{< note >}}
+There are similar [SetShaderOptionValue\_](#setshaderoptionvalue_-1) functions in the `ShaderItem` object that operate on a single shader.
+{{< /note >}}
+
+### GetShaderCount()
+
+Returns the number of shaders in the material type.
+
+### GetShader(`number`)
+
+Returns a [`ShaderItem`](#shaderitem-functions) at a given index, or a dummy [`ShaderItem`](#shaderitem-functions) if the index is out of bounds. See [ShaderItem functions](#shader-functions). 
+
+### GetShaderByTag(`string`)
+
+Returns a [`ShaderItem`](#shaderitem-functions) that has a given tag name, or a dummy [`ShaderItem`](#shaderitem-functions) if the name is not found. See [ShaderItem functions](#shader-functions) and [Shader tags in the Material Type File Specification](/docs/atom-guide/look-dev/materials/material-type-file-spec/#shaders). 
```

Comment:
Removed duplicate links. I think the one link suffices. Also fixed the link.
```suggestion
### GetShader(`number`)

Returns a `ShaderItem` at a given index, or a dummy `ShaderItem` if the index is out of bounds. See [ShaderItem functions](#shader-functions). 

### GetShaderByTag(`string`)

Returns a `ShaderItem` that has a given tag name, or a dummy `ShaderItem` if the name is not found. See [ShaderItem functions](#shaderitem-functions) and [Shader tags in the Material Type File Specification](/docs/atom-guide/look-dev/materials/material-type-file-spec/#shaders). 
```

Reply:
I think you meant to link shaderitem-functions.
```suggestion
### GetShader(`number`)

Returns a `ShaderItem` at a given index, or a dummy `ShaderItem` if the index is out of bounds. See [ShaderItem functions](#shaderitem-functions). 

### GetShaderByTag(`string`)

Returns a `ShaderItem` that has a given tag name, or a dummy `ShaderItem` if the name is not found. See [ShaderItem functions](#shaderitem-functions) and [Shader tags in the Material Type File Specification](/docs/atom-guide/look-dev/materials/material-type-file-spec/#shaders). 
```